File: 42. Trapping Rain Water/42.py
from typing import List

class Solution:
    def trap(self, height: List[int]) -> int:
        agua = 0
        mayorIzq = 0
        
        for index in range(len(height) - 1):

            if height[index] > mayorIzq:
                mayorIzq = height[index]

            mayorDer = max(height[index+1:]) 

            if (mayorIzq > height[index] and mayorDer > height[index]):

                menor = min(mayorIzq, mayorDer)

                agua += menor - height[index]
      
        return agua

File: 42. Trapping Rain Water/42.js
function waterTrap(height) {
  agua = 0

  for (let index = 0; index < height.length; index++) {

      let mayorIzq = Math.max(...height.slice(0, index));
      let mayorDer = Math.max(...height.slice(index + 1));

      if (mayorIzq > height[index] && mayorDer > height[index]) {

          menor = Math.min(mayorIzq, mayorDer)

          agua += menor - height[index]
      }
  }

  return agua;
};


module.exports = { waterTrap };
